cats_tutorial.ipynb

```diff
@@ -0,0 +1,491 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Simulating a smart thermostat using CATS: a Contextual Bandit algorithm with a continuous action space\n",
+    "\n",
+    "In this tutorial we will simulate the scenario of personalizing a thermostat for a household with two rooms, using Contextual Bandits in a continuous action space. The goal is to maximize user satisfaction with the thermostat quantified by measuring thermostat accuracy or reward (TR). The thermostat proposes a temperature and the user will either accept the temperature or adjust it to fit their needs.\n",
+    "\n",
+    "Let's recall that in a CB setting, a data point has four components,\n",
+    "\n",
+    "Context\n",
+    "Action\n",
+    "Probability of choosing action\n",
+    "Reward/cost for chosen action\n",
+    "\n",
+    "In our simulator, we will need to generate a context, get an action/decision for the given context and also simulate generating a reward.\n",
+    "\n",
+    "In our simulator, our goal is to maximize reward (TR) or minimize loss (-TR)\n",
```

Comment:
I would swap the sentence with this one: "The goal of the learning agent is to maximize the reward or to minimize the loss."

Reason for dropping TR and -TR: You already defined them above, so no need to redefine. You could use the abbreviation if you wish but I would wait to into -TR until after line 53 below